### PR TITLE
DIT-793: retrieving EORI details correctly

### PR DESCRIPTION
**Pull request by gokyo**

### CODE REVIEW

File: app/mapper/CaseRequestMapper.scala [hunk 1]
```diff
@@ -59,17 +42,17 @@ class CaseRequestMapper {
...
 
     val app = Application(
-      holder = holder,
+      holder = holderDetails,
       contact = contact,
       agent = agentDetails,
       offline = false,
-      goodName = describeYourItem.map(_.field1).getOrElse("CaseRequestMapperTestN/A"),
+      goodName = describeYourItem.map(_.field1).getOrElse("N/A"),
       goodDescription = describeYourItem.map(_.field2).getOrElse("N/A"),
```
Comment (gokyo) [inline app/mapper/CaseRequestMapper.scala]: Are we happy to save `N/A`?

File: app/mapper/CaseRequestMapper.scala [hunk 2]
```diff
@@ -59,17 +42,17 @@ class CaseRequestMapper {
...
+    val agentDetails: Option[AgentDetails] = buildAgentDetails(userEori)(answers)
+    val holderDetails: EORIDetails = buildHolderDetails(userEori)(answers)
 
     val app = Application(
-      holder = holder,
+      holder = holderDetails,
       contact = contact,
       agent = agentDetails,
       offline = false,
-      goodName = describeYourItem.map(_.field1).getOrElse("CaseRequestMapperTestN/A"),
```
Comment (gokyo) [inline app/mapper/CaseRequestMapper.scala]: Are we happy to save `N/A`?

File: app/mapper/CaseRequestMapper.scala (hunk 1 above)
Comment (pllaurado) [reply app/mapper/CaseRequestMapper.scala]: Maybe we could be more restrictive there by throwing and exception or stopping the process

Comment (pllaurado) [reply app/mapper/CaseRequestMapper.scala]: Ultimately this is mandatory data that which is strictly necessary to finish the process

Comment (gokyo) [reply app/mapper/CaseRequestMapper.scala]: Ok. I will throw an error

**Pull request merged**